### PR TITLE
Add error identification in deserializer code

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR introduces some simple error identification by setting
appropriate variables.

Targets #467 partly, as it removes **some** `@todo`s from
`src/serialize/json/deserializer_callbacks.c`.

### CODE REVIEW

File: src/serialize/json/deserializer.c [hunk 1]
```diff
@@ -192,6 +192,18 @@ deserialize(
...
     yajl_status stat = yajl_parse(d->handle, buffer, nbuf);
 
+    if (stat == yajl_status_client_canceled) {
+        // the parsing resulted in an error.
+        char* fmt;
+        if (d->error.parser_error) {
+            fmt = "YAJL parser error (code %i)";
+        } else {
+            fmt = "Deserializer error (code %i)";
+        }
```
Comment (manuelmessner) [inline src/serialize/json/deserializer.c:202]: I think, this should probably two static variables and a pointer to one of them.
I'm not quite sure whether the compiler will build the executable by using one string location and always pointing to the same offset or whether the string will be created multiple times.

**Comment by neithernut** [comment]
I smell collisions with other PRs.

**Comment by matthiasbeyer** [comment]
Do you mean merge collision? Well,... then I would need to rebase... obviousely. So I don't care until github tells me that it cannot be merged...

**Comment by neithernut** [comment]
Not merge collissions but semantical collisions. E.g. you were introducing new `@todo`s in the PR where you changes the definition of some method for `ws_string`.

**Comment by matthiasbeyer** [comment]
I'm ready here btw... feel free to review!

**Comment by neithernut** [comment]
ACK.

**Pull request merged**